### PR TITLE
cluster: fix the bug that region statistics are not updated after flow-round-by-digit change

### LINKED ISSUE #4295: After changing `flow-round-by-digit`, the region statistics are not updated
### What did you do?

* start a TiDB cluster
* run some tasks
* set PD config `flow-round-by-digit = 127`
* stop all tasks and wait for a while
* set PD config `flow-round-by-digit = 3`
* observe the `Key Visualizer` in `TiDB Dashboard`

### What did you expect to see?

After stopping the task, Key Visualizer should show no flow.

### What did you see instead?

image

### What version of PD are you using (`pd-server -V`)?

v5.2.2

### PR DESCRIPTION
**Pull request by HunDunDM**
### What problem does this PR solve?

fix #4295 

### What is changed and how it works?

A change in `flowRoundDivisor` also considers that the flow statistics have changed.

### Check List

Tests

- Unit test

Code changes

Side effects

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
Fix the bug that region statistics are not updated after `flow-round-by-digit` change.
```

### CODE REVIEW

File: server/core/region.go [hunk 1]
```diff
@@ -567,7 +567,8 @@ func GenerateRegionGuideFunc(enableLog bool) RegionGuideFunc {
 			// Once flow has changed, will update the cache.
 			// Because keys and bytes are strongly related, only bytes are judged.
 			if region.GetRoundBytesWritten() != origin.GetRoundBytesWritten() ||
-				region.GetRoundBytesRead() != origin.GetRoundBytesRead() {
+				region.GetRoundBytesRead() != origin.GetRoundBytesRead() ||
+				region.flowRoundDivisor != origin.flowRoundDivisor {
```
Comment (nolouch) [inline server/core/region.go:571]: How about `region.flowRoundDivisor < origin.flowRoundDivisor`?  To reduce unnecessary update.

File: server/grpc_service.go [hunk 2]
```diff
@@ -750,11 +750,11 @@ func (s *Server) RegionHeartbeat(stream pdpb.PD_RegionHeartbeatServer) error {
 			regionHeartbeatCounter.WithLabelValues(storeAddress, storeLabel, "report", "bind").Inc()
 			s.hbStreams.BindStream(storeID, server)
 			// refresh FlowRoundByDigit
-			FlowRoundByDigit = s.persistOptions.GetPDServerConfig().FlowRoundByDigit
+			regionFlowRoundByDigitOption = core.WithFlowRoundByDigit(s.persistOptions.GetPDServerConfig().FlowRoundByDigit)
```
Comment (rleungx) [inline server/grpc_service.go:753]: make it shorter?

File: server/core/region_option.go [hunk 3]
```diff
@@ -39,8 +39,9 @@ func WithDownPeers(downPeers []*pdpb.PeerStats) RegionCreateOption {
 
 // WithFlowRoundByDigit set the digit, which use to round to the nearest number
 func WithFlowRoundByDigit(digit int) RegionCreateOption {
+	flowRoundDivisor := uint64(math.Pow10(digit))
```
Comment (rleungx) [inline server/core/region_option.go:42]: no need?

Comment (HunDunDM) [reply server/core/region_option.go:42]: Reduce the number of `math.Pow10`.

File: server/grpc_service.go (hunk 2 above)
Comment (HunDunDM) [reply server/grpc_service.go]: rename to `flowRoundOption`

File: server/core/region_option.go (hunk 3 above)
Comment (HunDunDM) [reply server/core/region_option.go:42]: Now the return value (as func) of WithFlowRoundByDigit will be called a lot.

**Review by nolouch — APPROVED** [review]
lgtm

**Review by rleungx — APPROVED** [review]

**Pull request merged**